Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h

```diff
@@ -36,126 +36,393 @@
 
-// All permissions currently allowed, including non-expired granted permissions.
+/*!
+ @abstract The currently granted permissions that have not expired.
+ @seealso grantedPermissions
+ */
 @property (nonatomic, readonly, copy) NSSet<_WKWebExtensionPermission> *currentPermissions;
+
+/*!
+ @abstract The currently granted permission match patterns that have not expired.
+ @seealso grantedPermissionMatchPatterns
+ */
 @property (nonatomic, readonly, copy) NSSet<_WKWebExtensionMatchPattern *> *currentPermissionMatchPatterns;
 
+/*!
+ @abstract Checks the specified permission against the currently granted permissions.
+ @seealso currentPermissions
+ @seealso hasPermission:inTab:
+ @seealso permissionStateForPermission:
+ @seealso permissionStateForPermission:inTab:
+*/
 - (BOOL)hasPermission:(_WKWebExtensionPermission)permission;
+
+/*!
+ @abstract Checks the specified permission against the currently granted permissions in a specific tab.
+ @discussion Permissions can be granted on a per-tab basis. When the tab is known, permission checks should always use this method.
+ @seealso currentPermissions
+ @seealso hasPermission:
+ @seealso permissionStateForPermission:
+ @seealso permissionStateForPermission:inTab:
+ */
 - (BOOL)hasPermission:(_WKWebExtensionPermission)permission inTab:(nullable id <_WKWebExtensionTab>)tab;
 
+/*!
+ @abstract Checks the specified URL against the currently granted permission match patterns.
+ @seealso currentPermissionMatchPatterns
+ @seealso hasAccessToURL:inTab:
+ @seealso permissionStateForURL:
+ @seealso permissionStateForURL:inTab:
+ @seealso permissionStateForMatchPattern:
+ @seealso permissionStateForMatchPattern:inTab:
+ */
 - (BOOL)hasAccessToURL:(NSURL *)url;
+
+/*!
+ @abstract Checks the specified URL against the currently granted permission match patterns in a specific tab.
+ @discussion Some match patterns can be granted on a per-tab basis. When the tab is known, access checks should always use this method.
+ @seealso currentPermissionMatchPatterns
+ @seealso hasAccessToURL:
+ @seealso permissionStateForURL:
+ @seealso permissionStateForURL:inTab:
+ @seealso permissionStateForMatchPattern:
+ @seealso permissionStateForMatchPattern:inTab:
+ */
 - (BOOL)hasAccessToURL:(NSURL *)url inTab:(nullable id <_WKWebExtensionTab>)tab;
 
-// Specifically checks for any <all_urls> patterns. It does not check for all hosts patterns.
-// This is needed for APIs like tabs.captureVisibleTab that need to check specifically for <all_urls>.
-// You likely want to use hasPermissionToAccessAllHosts instead.
+/*!
+ @abstract Checks if the currently granted permission match patterns set contains the `<all_urls>` pattern.
+ @discussion This does not check for any `*` host patterns. In most cases you should use the broader `hasAccessToAllHosts`.
+ This check is primarily needed for APIs like `browser.tabs.captureVisibleTab()` that need to check specifically for `<all_urls>`.
```

Comment:
Are there any other APIs that use this check, or is it truly only for `captureVisibleTab`?

Reply:
IIRC it is only `captureVisibleTab` but clients could use this for permission warning text as they see fit.